Kudu.Core/Deployment/Generator/ExternalCommandBuilder.cs

```diff
@@ -85,5 +114,94 @@ protected void RunCommand(DeploymentContext context, string command)
                 throw;
             }
         }
+
+        /// <summary>
+        /// Mark public for testing
+        /// </summary>
+        public List<string> GetPostBuildActionScripts()
+        {
+            List<string> scriptFilesGroupedAndSorted = new List<string>();
+            List<string> scriptFolders = new List<string>();
+
+            // "/site/deployments/tools/PostDeploymentActions" (can override with %POST_DEPLOYMENT_ACTIONS_DIR%)
+            var defaultPath = Path.Combine(Environment.DeploymentToolsPath, PostDeploymentActions);
+            if (!string.IsNullOrEmpty(System.Environment.GetEnvironmentVariable("POST_DEPLOYMENT_ACTIONS_DIR")))
+            {
+                defaultPath = System.Environment.GetEnvironmentVariable("POST_DEPLOYMENT_ACTIONS_DIR");
+            }
+
+            if (FileSystemHelpers.DirectoryExists(defaultPath))
+            {
+                scriptFolders.Add(defaultPath);
+            }
+
+            // D:\Program Files (x86)\SiteExtensions
+            string siteExtensionFolder = Path.Combine(System.Environment.GetFolderPath(System.Environment.SpecialFolder.ProgramFilesX86), "SiteExtensions");
+            // Find all enviroment variable with format {package id}_EXTENSION_VERSION,
+            // And find scripts from PostDeploymentActions" folder
+            foreach (DictionaryEntry entry in System.Environment.GetEnvironmentVariables())
+            {
+                string key = entry.Key.ToString();
+                string val = entry.Value.ToString();
+                NuGetVersion ver = null;
+                if (key.EndsWith("_EXTENSION_VERSION", StringComparison.Ordinal)
```

Comment:
Should be `OrdinalIgnoreCase`. We never treat env variable names as case sensitive.